Public/Src/Cache/ContentStore/Distributed/NuCache/EventStreaming/AzureEventHubClient.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Tracing;
+using BuildXL.Cache.ContentStore.Tracing.Internal;
+using BuildXL.Cache.ContentStore.Utils;
+using BuildXL.Utilities.Tasks;
+using Microsoft.Azure.EventHubs;
+
+namespace BuildXL.Cache.ContentStore.Distributed.NuCache.EventStreaming
+{
+    /// <summary>
+    /// An event hub client which interacts with Azure Event Hub service
+    /// </summary>
+    public class AzureEventHubClient : StartupShutdownSlimBase, IEventHubClient
+    {
+        private const string PartitionId = "0";
+
+        private EventHubClient _eventHubClient;
+        private PartitionSender _partitionSender;
+        private readonly EventHubContentLocationEventStoreConfiguration _configuration;
+
+        private readonly string _hostName = Guid.NewGuid().ToString();
+
+        private PartitionReceiver _partitionReceiver;
+
+        protected override Tracer Tracer { get; } = new Tracer(nameof(AzureEventHubClient));
+
+        /// <nodoc />
+        public AzureEventHubClient(EventHubContentLocationEventStoreConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        /// <inheritdoc />
+        public BoolResult StartProcessing(OperationContext context, EventSequencePoint sequencePoint, IPartitionReceiveHandler processor)
+        {
+            Tracer.Info(context, $"{Tracer.Name}: Initializing event processing for event hub '{_configuration.EventHubName}' and consumer group '{_configuration.ConsumerGroupName}'.");
+
+            if (_partitionReceiver == null)
+            {
+                _partitionReceiver = _eventHubClient.CreateReceiver(
+                    _configuration.ConsumerGroupName,
+                    PartitionId,
+                    GetInitialOffset(context, sequencePoint),
+                    new ReceiverOptions()
+                    {
+                        Identifier = _hostName
+                    });
+
+                _partitionReceiver.SetReceiveHandler(processor);
+            }
+
+            return BoolResult.Success;
+        }
+
+        /// <inheritdoc />
+        public BoolResult SuspendProcessing(OperationContext context)
+        {
+            // In unit tests, hangs sometimes occur for this when running multiple tests in sequence.
+            // Adding a timeout to detect when this occurs
+            _partitionReceiver?.CloseAsync().WithTimeoutAsync(TimeSpan.FromMinutes(1)).GetAwaiter().GetResult();

```

Comment:
if `_partitionReceiver` is null this code will fail with NRE.

I think you have to wrap it into an if block.